src/amp-story-embed.js

```diff
@@ -123,34 +194,145 @@ export class AmpStoryEmbed {
   }
 
   /**
-   * @return {!Promise}
    * @public
    */
   layoutCallback() {
-    // TODO: Layout all child iframes.
-    return this.layoutIframe_(0);
+    if (this.isLaidOut_) {
+      return;
+    }
+
+    const iframes = toArray(this.rootEl_.querySelectorAll('iframe'));
```

Comment:
As you create/recycle iframes as a user progresses through the stories, you won't be able to do `querySelectorAll` every time you want an iframe.
You should keep references to the iframes like you keep references to the `Messaging`. This will potentially change your data structure choice (`messagingFor`), so I believe this should be done in this PR.